src/crawlee/events/_types.py

```diff
@@ -59,6 +59,10 @@ class EventMigratingData(BaseModel):
 
     model_config = ConfigDict(populate_by_name=True)
 
+    # The remaining time in seconds before the migration is forced and the process is killed
+    # Optional because it's not present when the event handler is called manually
+    time_remaining_secs: Annotated[float | None, Field(alias='timeRemainingSecs')]
```

Comment:
Could we use `datetime`/`time` types rather than float + secs suffix identifier?

Reply:
But if you want it as `timedelta` in Python, I can add it, it shouldn't be hard. Only the property names will be slightly different between JS and Python, but I guess it's fine.

Reply:
Yes please. We use `timedelta` for all time-related fields. For Pydantic models, we have a custom type called [timedelta_ms](https://github.com/apify/crawlee-python/blob/master/src/crawlee/_utils/models.py#L38), which includes a custom validator and serializer. You can see it in use e.g. [here](https://github.com/apify/crawlee-python/blob/ae0d328c85cc98a962b320ad876342269157acca/src/crawlee/statistics/_models.py#L68).

And the property will only differ in Python’s internal code, the env var name can remain unchanged.

Reply:
It's not an env var, it's data in an Actor event, and it's not internal, since it's something that users will be working with.

I'll rename the field exposed to users to `time_remaining`, which will be a `timedelta`, and internally it will be parsed from the `timeRemainingSecs` property of the received event data.